### PR TITLE
fix: 🤔 add z-index tokens 

### LINKED ISSUE #521: fix: 🤔 add z-index tokens
## Current behavior
I've noticed, thanks to a user's feedback on the Solid components, that the z-index variable we currently have in the sd-dropdown component is not functioning as expected. When I inspect the HTML in the browser, I also observe that I can find the values of the other variables by clicking, but var(--sd-z-index-dropdown) is not clickable.
image
image

This creates the issue where the dropdown's popup could end up behind another element.

## Expected behavior
Adding Sheolace tokens for z-index in our library
https://shoelace.style/tokens/z-index

## DoR
- [x] Item has business value
- [ ] Item has been estimated by the team
- [x] Item is clear and well-defined
- [x] Item dependencies have been identified

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch

### PR DESCRIPTION
**Pull request by Vahid1919**
## Description:
Added z-index tokens for tailwind. Availability as CSS variables is still a WIP .Closes #521

## Definition of Reviewable:
*PR notes: Irrelevant elements should be removed.*
- [x] relevant tickets are linked
- [x] PR is assigned to project board

### CODE REVIEW

**Comment by Vahid1919** [comment]
@mariohamann @yoezlem 
I managed to get the tokens to work in tailwind.
image


 **But** they still don't work when used directly as css variables. So for cases like a drop down, we have two options: (1) Set a fall back value. 
image

(2) Figure out how to access the variables I just set in the tokens. 

Which one should I do? If we pick (2), any hints on how to approach this would be appreciated :)

File: packages/tokens/src/create-theme.cjs [hunk 1]
```diff
@@ -245,6 +245,17 @@ const getCoreTokensByType = (tokenType, cssVariableScope) => {
       })${description ? ` /* ${description} */` : ''}`;
     }
   });
+  return result;
+};
+
+const getZIndices = () => {
```
Comment (yoezlem) [inline packages/tokens/src/create-theme.cjs:251]: @mariohamann @coraliefeil Do we need here some logic for which "settings/components" are open for z-index?

**Comment by yoezlem** [comment]
> @mariohamann @yoezlem I managed to get the tokens to work in tailwind. image
> 
> **But** they still don't work when used directly as css variables. So for cases like a drop down, we have two options: (1) Set a fall back value. image
> 
> (2) Figure out how to access the variables I just set in the tokens.
> 
> Which one should I do? If we pick (2), any hints on how to approach this would be appreciated :)

Question for @mariohamann 
Would like to have approach 2. But I have no clue why its not working out of the box..🤔

**Comment by Vahid1919** [comment]
@solid-design-system/development Switching this into a regular PR since I am finished with the changes on my side. Please review when you get the chance :)

**Pull request merged**